src/main/java/me/moodcat/api/SongAPI.java

```diff
@@ -119,16 +129,44 @@ public SongModel getSongById(@PathParam("id") final int id) {
     public ClassificationRequest classifySong(@PathParam("id") final int id,
             final ClassificationRequest classification)
             throws InvalidClassificationException {
+
+        final User user = this.currentUserProvider.get();
         final Song song = this.songDAO.findBySoundCloudId(id);
+        final VAVector classificationVector = new VAVector(
+            classification.getValence(),
+            classification.getArousal());
+
+        if(classificationDAO.exists(user, song)) {
+            /*
+             * This should not happen, because after the initial classification
+             * the song will not be available for classification (and this endpoint
+             * should throw an IllegalArgumentException instead). However, this
+             * endpoint is currently also abused for the classification game.
+             *
+             * In order to prevent users to abuse this to gain points with fake
+             * classifications on the same song, we ignore duplicate classifications.
```

Comment:
(and the frontend issue is in no way an easy fix, it requires quite a large overhaul and we need this fixed fast)